src/script/deploy.js

```diff
@@ -0,0 +1,283 @@
+
+async function getEthereumEtherscanApiKey(config) {
+  if (!config.ethereumEtherscanApiKey) {
+    config.ethereumEtherscanApiKey = process.env.ETHERSCAN_API_KEY;
+  }
+}
+
+async function getLineaEtherscanApiKey(config) {
+  if (!config.lineaScanApiKey) {
+    config.lineaScanApiKey = process.env.LINEA_SCAN_API_KEY;
+  }
+}
+
+async function getTreeDepth(config) {
+  if (!config.treeDepth) {
+    config.treeDepth = await ask("Enter WorldID tree depth: ");
+  }
+}
+
+async function getLineaWorldIDAddress(config) {
+  if (!config.lineaWorldIDAddress) {
+    config.lineaWorldIDAddress = process.env.LINEA_WORLD_ID_ADDRESS;
+  }
+}
+
+async function getLineaStateBridgeAddress(config) {
+  if (!config.lineaStateBridgeAddress) {
+    config.lineaStateBridgeAddress = process.env.LINEA_STATE_BRIDGE_ADDRESS;
+  }
+}
+
+async function getWorldIDIdentityManagerAddress(config) {
+  if (!config.worldIDIdentityManagerAddress) {
+    config.worldIDIdentityManagerAddress = process.env.WORLD_ID_IDENTITY_MANAGER_ADDRESS;
+  }
+}
+
+///////////////////////////////////////////////////////////////////
+///                            UTILS                            ///
+///////////////////////////////////////////////////////////////////
+async function loadConfiguration(useConfig) {
+  if (!useConfig) {
+    return {};
+  }
+  let answer = await ask(`Do you want to load configuration from prior runs? [Y/n]: `, "bool");
+  const spinner = ora("Configuration Loading").start();
+  if (answer === undefined) {
+    answer = true;
+  }
+  if (answer) {
+    if (!fs.existsSync(CONFIG_FILENAME)) {
+      spinner.warn("Configuration load requested but no configuration available: continuing");
+      return {};
+    }
+    try {
+      const fileContents = JSON.parse(fs.readFileSync(CONFIG_FILENAME).toString());
+      if (fileContents) {
+        spinner.succeed("Configuration loaded");
+        return fileContents;
+      } else {
+        spinner.warn("Unable to parse configuration: deleting and continuing");
+        fs.rmSync(CONFIG_FILENAME);
+        return {};
+      }
+    } catch {
+      spinner.warn("Unable to parse configuration: deleting and continuing");
+      fs.rmSync(CONFIG_FILENAME);
+      return {};
+    }
+  } else {
+    spinner.succeed("Configuration not loaded");
+    return {};
+  }
+}
+
+async function saveConfiguration(config) {
+  const oldData = (() => {
+    try {
+      return JSON.parse(fs.readFileSync(CONFIG_FILENAME).toString());
+    } catch {
+      return {};
+    }
+  })();
+
+  const data = JSON.stringify({ ...oldData, ...config });
+  fs.writeFileSync(CONFIG_FILENAME, data);
+}
+
+///////////////////////////////////////////////////////////////////
+///                         DEPLOYMENTS                         ///
+///////////////////////////////////////////////////////////////////
+
+async function deployLineaWorldID(config) {
+  const spinner = ora("Deploying LineaID on Linea...").start();
+
+  try {
+    const data = execSync(
```

Comment:
I think we may need to save the address of newly deployed `LineaWorldID` to pass it to the `LineaStateBridge` constructor